common/djangoapps/util/date_utils.py

```diff
@@ -10,9 +10,46 @@
 import six
 from django.utils.translation import pgettext, ugettext
 from pytz import UnknownTimeZoneError, timezone, utc
+from xmodule.course_module import DEFAULT_START_DATE
+from openedx.core.djangoapps.schedules.models import Schedule
 
 
-def get_default_time_display(dtime):
+def is_unset(date_field_value):
+    if isinstance(date_field_value, timedelta):
+        return date_field_value.total_seconds() <= 0
+    else:
+        return date_field_value == DEFAULT_START_DATE
+
+
+def course_absolute_time(course, date_field_value):
+    """
+    Return an absolute time for a given course and date_field_value.
+    The date_field_value may be absolute or relative.
+    """
+    if isinstance(date_field_value, timedelta):
+        return course.start + date_field_value
+    else:
+        return date_field_value
+
+
+def user_absolute_time(user, course_key, date_field_value):
+    # This is a delayed import to allow the CourseDescriptor to use course_absolute_time
+
+    from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+    if isinstance(date_field_value, timedelta):
+        if user.is_authenticated:
+            try:
+                course_start = user.courseenrollment_set.get(course_id=course_key).schedule.start
+            except Schedule.DoesNotExist:
+                course_start = user.courseenrollment_set.get(course_id=course_key).course.start
+        else:
+            course_start = CourseOverview.get_from_id(course_key).start
+        return course_start + date_field_value
+    else:
+        return date_field_value
+
+
+def get_default_time_display(time_to_display):
     """
     Converts a datetime to a string representation. This is the default
```

Comment:
maybe update this docstring to mention the 2 supported input types and differing output formats